pkg/kubelet/volume_host.go

```diff
@@ -190,6 +206,15 @@ func (kvh *kubeletVolumeHost) WaitForCacheSync() error {
 		return fmt.Errorf("failed to wait for cache sync for CSIDriverLister")
 	}
 
+	if kvh.csiNodeSynced != nil {
```

Comment:
I think we need to modify the `csiDriversSynced` check on L198 to handle the case that the csiDriver and csiNode feature gates can be toggled independently